Userland/Libraries/LibC/string.cpp

```diff
@@ -141,32 +141,7 @@ void* memcpy(void* dest_ptr, void const* src_ptr, size_t n)
 #endif
 }
 
-#if ARCH(I386)
-// https://pubs.opengroup.org/onlinepubs/9699919799/functions/memset.html
-void* memset(void* dest_ptr, int c, size_t n)
-{
-    size_t dest = (size_t)dest_ptr;
-    // FIXME: Support starting at an unaligned address.
-    if (!(dest & 0x3) && n >= 12) {
-        size_t size_ts = n / sizeof(size_t);
-        size_t expanded_c = explode_byte((u8)c);
-        asm volatile(
-            "rep stosl\n"
-            : "=D"(dest)
-            : "D"(dest), "c"(size_ts), "a"(expanded_c)
-            : "memory");
-        n -= size_ts * sizeof(size_t);
-        if (n == 0)
-            return dest_ptr;
-    }
-    asm volatile(
-        "rep stosb\n"
-        : "=D"(dest), "=c"(n)
-        : "0"(dest), "1"(n), "a"(c)
-        : "memory");
-    return dest_ptr;
-}
-#elif ARCH(X86_64)
+#if ARCH(X86_64)
```

Comment:
This comment and check are confusing without the i686 implementation above.

A comment explanining that "this is where I would put my memset implementation if I had one", with the posix link is probably better

Reply:
Got it, thanks :)